File: mario/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int j ;
    do
    {
        j = get_int("Height: ");
    }
    while (j <= 0 || j > 8); //retrieves user input

    for (int i = 0; i < j; i++) //loops through input to generat piramid
    {

        if (j > 1) //generates left spaces
        {
            for (int p = 1; p <= j - i - 1; p++)
            {
                printf(" ");
            }
        }

        for (int x = j - i - 1 ; x < j; x++) //generates left #'s
        {
            printf("#");
        }

        printf("  "); //generates middle space

        for (int x = j - i - 1 ; x < j; x++) //generates right #'s
        {
            printf("#");
        }

        printf("\n"); //goes to next line
    }
}

File: mario.py/mario.py
from cs50 import get_int


def main(h):

    # Asks for input while h is smaller or equal to 0 or bigger then 8
    while h <= 0 or h > 8:
        h = get_int("Height: ")

    # determine width
    w = h

    # create piramid
    for i in range(1, h+1):
        space = w - i
        print(" " * space, end="")
        print("#" * i, end="")

        print("  ", end="")

        print("#" * i)


if __name__ == "__main__":
    main(0)